test/api_test.js

```diff
@@ -569,6 +622,105 @@ describe('Api', () => {
     });
   });
 
+  describe('credentials', () => {
+    const P = f => path.resolve(__dirname, f);
+
+    const loadFile = (filename) => {
+      return fs.readFileSync(P(filename)).toString('base64');
+    };
+
+    const captureError = async(fn, args) => {
+      try {
+        await fn.apply(client, args);
+        return 'Should fail';
+      } catch (err) {
+        return err.code;
+      }
+    };
+
+    const expectedError = 'AuthorizationFailed';
+    const createTemporaryCredentials = taskcluster.createTemporaryCredentials;
+
+    before(() => {
+      taskcluster.createTemporaryCredentials = sandbox.stub();
+      taskcluster.createTemporaryCredentials.returns({
```

Comment:
If we set a variable `expectedCredentials`, we could use that to set the return value of this stub as well as using it to ensure that the credentials to test against would be using the same object with `assume(actualCredentials).deeply.equals(expectedCredentials)`